### ISSUE TITLE
s3 sources: Support protobuf-encoded objects

### ISSUE THREAD

**Issue by quodlibetor** [MaterializeInc/materialize#5597]
Currently S3 sources only support text-like encodings. There are a few reasons for this:

* We need a way to specify arbitrary record delimiters (#2461)
* Protobuf needs a schema to be parseable
* S3 sources need to be able to stream byte chunks through to decoders, but byte chunks might not be complete if you're not using a length-prefixed record delimiter, and decoder's currently expect each message to be an entire record.

Basically the same as #5596 , but for protobuf.

**Comment by sploiselle** [comment]
we deprecated S3 sources
